torchbenchmark/models/pyhpc_equation_of_state/eos_pytorch.py

```diff
@@ -0,0 +1,199 @@
+"""
+==========================================================================
+  in-situ density, dynamic enthalpy and derivatives
+  from Absolute Salinity and Conservative
+  Temperature, using the computationally-efficient 48-term expression for
+  density in terms of SA, CT and p (IOC et al., 2010).
+==========================================================================
+"""
+
+import numpy as np
+import torch
+
+
+@torch.jit.script
+def gsw_dHdT(sa, ct, p):
```

Comment:
Why don't these guys use autograd? 🤔 I guess in this case they need an AOT autograd to make it run reasonably fast.

Reply:
good lord. i would love to have to code-review that formula..

Reply:
@wconstab  @bertmaher  do we want to create a new category for this type of benchmarks?

Reply:
> @wconstab @bertmaher do we want to create a new category for this type of benchmarks?

Sure!  It's like HPC, or "scientific computing"